### ISSUE TITLE
add documentation on performing a review

### ISSUE THREAD

**Issue by capsulecorplab** [tetrabiodistributed/tetra#37]
Following #33, documentation should include a how-to guide for performing reviews on GitHub

**Comment by cyunis** [comment]
i have a shortlist for doing this, trialing with @thehannahfink this week

**Comment by cyunis** [comment]
Suggestion for Change management page addition: 

Lifecycle of an Issue/Pull Request:
1) Go to the repo and make a branch titled name-of-issue-# where # is the issue number.
2) Now make your edits. We are working in Asciidocs (https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/). Commit to the branch.
3) If you want to view your edits you need to clone locally and view in Docker. If you’ve already cloned first checkout the main branch, pull and then do git checkout -b name-of-new-branch-#<issue no.> to create a local branch. Push the edits for specific files you’ve worked on.
4) Create a pull request (PR) titled “Closes #: name of issue” where # is the issue number. Include multiple issue numbers if the PR closes multiple issues. Convert the PR to a draft until you are done editing, then mark ready for a review.
5) Review the changes - provide feedback on Github (you should get an email notification). If it looks good then approve. If not, give feedback and notify the person who needs to review changes.

(3) may need subbullets or a separate section for handling pulling and pushing while on cloned branches.

**Comment by capsulecorplab** [comment]
> Suggestion for Change management page addition:
> 
> Lifecycle of an Issue/Pull Request:
> 
>     1. Go to the repo and make a branch titled name-of-issue-# where # is the issue number.
> 
>     2. Now make your edits. We are working in Asciidocs (https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/). Commit to the branch.
> 
>     3. If you want to view your edits you need to clone locally and view in Docker. If you’ve already cloned first checkout the main branch, pull and then do git checkout -b name-of-new-branch-# to create a local branch. Push the edits for specific files you’ve worked on.
> 
>     4. Create a pull request (PR) titled “Closes #: name of issue” where # is the issue number. Include multiple issue numbers if the PR closes multiple issues. Convert the PR to a draft until you are done editing, then mark ready for a review.
> 
>     5. Review the changes - provide feedback on Github (you should get an email notification). If it looks good then approve. If not, give feedback and notify the person who needs to review changes.
> 
> 
> (3) may need subbullets or a separate section for handling pulling and pushing while on cloned branches.

That documentation already exists. See https://tetrabiodistributed.github.io/project-tetra-docs/how-to-guides/contribution-guidelines/

**Comment by cyunis** [comment]
i will look at website to double check overlap.
i think that most issues will be resolved by changing titles (aka #41 )

**Comment by Burhan-Q** [comment]
[Draft of GitHub process for Hardware](https://github.com/tetrabiodistributed/project-tetra-docs/blob/hardware_procedures/content/en/how-to-guides/internal-guides/Hardware_design_and_GitHub.md)

Feel free to provide feedback on the document in the comments of this issue

**Comment by cyunis** [comment]
@Burhan-Q comments (you wrote a lot, so i have a lot of comments in response): In general this documentation reads a lot more like a tutorial than a how to guide - although the review section is closer to how to and there is some ordering of steps overall. Choosing to write this documentation as tutorials or how to guide exclusively might tighten the flow of how it reads. I have suggestions for how to split this up since I think you can make documentation for BOTH tutorials and how to from this draft. I just want my comments here to be focused on what you've already written. Also I HIGHLY recommend reading out loud when you revise, it helps catch weird wording and run on sentences. Now for more detailed feedback...

-I don't think the numbering is adding much to the content other than providing a break between lines. Like, the steps don't build on each other for each step to make a progression of tasks because some steps seem to be able to be performed in parallel. I would consider dropping the numbering for paragraphs and using bullet points if you want to make lists (for explanations) or keep the numbering but trying to separate steps in the order of progression (for how tos).
-I would shift your writing tone from passive to active if you choose to make this how to documentation. It makes the flow more like a to do list - more action oriented (for example: "1. When a design change, addition or modification is made, create a Github Issue (link to section describing how to make issue)."
-Instead of linking out to the definition of done sheet I would put the requirements in a bullet point list. Saves the trouble of moving to a separate page and also that sheet is publicly editable rn. Also the design changes are currently listed here so maybe you could reference this list/add on to it for design task def'ns instead of rewriting? https://tetrabiodistributed.github.io/project-tetra-docs/how-to-guides/internal-guides/change_management/#_how_to_make_design_changes
-Do you want to maybe take the expectations in the steps and put them in a separate blurb of their own before the to do list?
-For step 3 in github issues is there anything special to this other than assigning a task on github?
-"whoever is last assigned" what does this mean? (step 2 of assignments)
-is there a need to comment on the role of the assignee on an issue? maybe i'm unaware of something on hardware but it seems like the role is mostly to get the work done and/or provide feedback so is there any more specific distinction?
-in github labels section you have a typo in #2 "Currently the some key labels"
-have we done comments on reasons for label changes? that seems like it might be an extra step of commenting without much added value.
-i would love to hear feedback from the people designing hardware on #2 in the review process. 
-this review section is the closest to a how to from all thats written so far
-in #5b what did you have in mind in terms of responsibilities?
-how much do we need to explain how to use the features on github in our documentation? (thinking about step 6b when youre talking about who the assignee needs to be changed to/from)
-#7 might be easier to say "in Step 3 from Assignments" and hyperlink to the assignments section (at least thats the format we're using for other references so far)
-is "scope of work" in 7a referring to scope creep?
-why isn't bundling of issues discussed as part of the review section?
-in the pull requests section i'm not quite sure what #4 is describing. are you saying there should be "Issue #444: closed" written somewhere in the PR?
-lol proper protocol we fancy. tbh i dont know if anyone other than @capsulecorplab is doing merging right now?
-also the labeling section is super helpful for me i have been wondering what those labels mean. love it.
-github milestones reads like an explanation to me almost

lastly just want to drop two links that might help:
https://documentation.divio.com/how-to-guides/#how-to-write-good-how-to-guides
https://documentation.divio.com/tutorials/
